### ISSUE TITLE
Flipping from one instance to another while in details panel causes scope to keep loading details

### ISSUE THREAD

**Issue by abuehrle** [weaveworks/scope#2015]
When viewing the fluxd agent in one instance and then switching to another 'empty' instance, the new instance tried loading the detail panel from the instance that I was viewing.  (perhaps it shouldn't even know about that other instance and shouldn't try to load the details.)

![screen shot 2016-11-15 at 1 41 36 pm](https://cloud.githubusercontent.com/assets/13420383/20322983/9e730880-ab40-11e6-9136-b3b1a3fc7856.png)

**Comment by 2opremio** [comment]
What do you mean by instance. Weave Cloud instance?

**Comment by 2opremio** [comment]
It would probably be best if you could get a video. I would suggest using http://recordit.co/

**Comment by abuehrle** [comment]
I mean cloud instances.  To recreate: 

1. Deployed flux agent to one Weave cloud instance in my kube cluster.
2. Clicked on Flux Agent and viewed details panel and then drilled down to container level from pod.
3. Switched contexts by going to a different instance I have that is empty, i.e. no probes deployed.
4. Cloud showed me the details panel of the other instance as shown above and kept trying to load it.

**Comment by 2opremio** [comment]
@foot / @davkal Mind taking a look to decide whether this should be moved to service-ui ?

**Comment by rade** [comment]
I thought this behaviour was by design, i.e. we wanted users to be able to switch instances while retaining context, and just because there is no data for a particular UI element at the moment, doesn't mean there won't be a few secs later.

Nevertheless, perhaps the "no data at all" case needs special casing.

**Comment by abuehrle** [comment]
I don't think it's the no data that's the problem. It's the activity monitor that appears which makes it seem that Scope is working to bring in some metrics, which aren't there, since it's in this wrong instance.

**Comment by rade** [comment]
Scope doesn't know it's in the "wrong instance"; as far as it is concerned you just switched between two instances and are attempting to look at the "same" container in both. The container may not be there right now in the 2nd instance, but it could, in theory, appear any second.

This is of course assuming that the details panel references things nominally, rather than by some identity which would never be the same across instances - @davkal @foot @2opremio is that actually the case? If not then details panels would never work across instances, in which case they should not be retained.

**Comment by foot** [comment]
True, details panel visibility is based on ID and so won't ever load onInstanceSwitch.

We can start flagging parts of the ui-state as "non-transferable"..

**Comment by foot** [comment]
This might be a bit of service-ui work and a bit of scope work depending on how its implemented. But its more a service issue in a sense.

**Comment by davkal** [comment]
Keeping the entity visible after the instance switch is desired goal. The current matching by ID makes it only work for some topologies, and probably useless for others (e.g., process ID).

I'd try a weaker field to match, e.g. node label and see how it feels. If it does not behave as intended, I would remove the details panel on instance switch.

**Comment by jpellizzari** [comment]
This may be easier to tackle if #2144 gets merged.

**Comment by fbarl** [comment]
@davkal I agree that node matching by ID between instances doesn't work at all for some topologies, but I also doubt we could come up with a sensible matching heuristics that would never match the wrong nodes (here I assume that a wrong match would be less desirable than not detecting a proper match, i.e. false positives being worse than false negatives).

To give an example from the *Containers* topology between our `dev` and `prod` instances, there is almost never a match by Container ID between the instances, but there are many containers with the same label across both instances, so that wouldn't be a reliable matching criterion. If we also used sublabels information, whenever they contained an IP, there would almost never going to be a match. I couldn't think of any other universal & reliable fields we could use for matching.

I think there might be some room for improvement if we focused on the special cases, but as a general solution, I'd prefer always removing the details panel after switching instances. @davkal, @foot, @2opremio thoughts?
